src/Resources/Sprite.cpp

```diff
@@ -571,88 +575,23 @@ void Sprite::addDefaultAction()
 	{
 		auto& imageSheet = mImageSheets["default"]; // Adds a default sheet. //-V607
 
-		int width = imageSheet.width();
-		int height = imageSheet.height();
+		const auto size = imageSheet.size();
+		const auto bounds = NAS2D::Rectangle<int>::Create(NAS2D::Point{0, 0}, size);
+		const auto anchorOffset = size / 2;
 
-		FrameList frameList{SpriteFrame("default", 0, 0, width, height, width / 2, height / 2, -1)};
+		FrameList frameList{SpriteFrame{"default", bounds, anchorOffset, -1}};
 		mActions["default"] = frameList;
 	}
 }
 
 
-/**
- * Gets the width of the Sprite.
- *
- * \note	This gets the width of the current frame. For most
- *			most sprites this will be the same for all frames
- *			but can be surprising if frame sizes vary.
- */
-int Sprite::width() const
+Vector<int> Sprite::size() const
 {
-	return mActions.at(mCurrentAction)[mCurrentFrame].width();
+	return mActions.at(mCurrentAction)[mCurrentFrame].bounds.size();
 }
 
 
-/**
- * Gets the height of the Sprite.
- *
- * \note	This gets the height of the current frame. For most
- *			most sprites this will be the same for all frames
- *			but can be surprising if frame sizes vary.
- */
-int Sprite::height() const
+Point<int> Sprite::origin(Point<int> point) const
 {
-	return mActions.at(mCurrentAction)[mCurrentFrame].height();
+	return point - mActions.at(mCurrentAction)[mCurrentFrame].anchor;
```

Comment:
This is a displacement rather than a point. Consider the function return datatype be changed to `Vector`.

Reply:
Again, that requires a new definition of `anchor`, or an updated name for it to make sense.

The name `origin` suggests the return type should be `Point`.